Confus/Confus/Flag.cpp

```diff
@@ -109,71 +117,100 @@ namespace Confus {
 		}
-		else if (a_PlayerObject->TeamIdentifier == *m_TeamIdentifier) {
+		else if (*a_PlayerObject->TeamIdentifier == *m_TeamIdentifier) 
+        {
 			//If flag has been dropped return flag to base
-			if (*m_FlagStatus == EFlagEnum::FLAG_DROPPED) {
-				m_FlagNode->setParent(NULL);
-				returnToStartPos();
+ 			if (*m_FlagStatus == EFlagEnum::FlagDropped) 
+            {
+                MyLogger->log("Returning flag.");
+                returnToStartPosition();
 			}
 			//If flag is at base and player is carrying a flag
-			else if (*m_FlagStatus == EFlagEnum::FLAG_BASE) {
-				if (a_PlayerObject->CarryingFlag == EFlagEnum::FLAG_TAKEN) {
-					score(a_PlayerObject);
+			else if (*m_FlagStatus == EFlagEnum::FlagBase) 
+            {
+				if (*a_PlayerObject->CarryingFlag == EFlagEnum::FlagTaken) 
+                {					
+                    if(a_PlayerObject->FlagPointer != nullptr) {
+                        MyLogger->log("Scoring flag.");
+                        a_PlayerObject->FlagPointer->returnToStartPosition();
+                        a_PlayerObject->FlagPointer = nullptr;
+                        score(a_PlayerObject);
+                    }
+                    else {
+                        MyLogger->log("Failed to get flag.");
+                    }
 				}
 			}
 		}
 	}
 
-	///Score points to team of a_PlayerObject
-	void Flag::score(Player* a_PlayerObject) {
-		
+	//TODO Score points to team of a_PlayerObject
+	void Flag::score(Player* a_PlayerObject) 
+    {
+        a_PlayerObject->CarryingFlag = new EFlagEnum(EFlagEnum::None);
 	}
 
-	///Drop Flag at position of player
-	void Flag::drop(Player* a_PlayerObject) {
-		
+	//TODO Drop Flag at position of player
+	void Flag::drop(Player* a_PlayerObject) 
+    {
+        m_FlagNode->setParent(m_FlagOldParent);
+        m_FlagNode->setPosition(a_PlayerObject->getAbsolutePosition());
+        a_PlayerObject->FlagPointer = nullptr;
+        m_FlagStatus = new EFlagEnum(EFlagEnum::FlagDropped);
+        a_PlayerObject->CarryingFlag = new EFlagEnum(EFlagEnum::None);
 	}
 
-    void Flag::setStartPosition(irr::core::vector3df a_Position) {
+    void Flag::setStartPosition(irr::core::vector3df a_Position) 
+    {
         m_StartPosition->set(a_Position);
     }
 
-    void Flag::setStartRotation(irr::core::vector3df a_Rotation) {
+    void Flag::setStartRotation(irr::core::vector3df a_Rotation) 
+    {
         m_StartRotation->set(a_Rotation);
     }
 
-    void Flag::returnToStartPos() {
+    void Flag::returnToStartPosition() {
+        m_FlagNode->setParent(m_FlagOldParent);
         m_FlagNode->setPosition(*m_StartPosition);
         m_FlagNode->setRotation(*m_StartRotation);
-		*m_FlagStatus = EFlagEnum::FLAG_BASE;
+		*m_FlagStatus = EFlagEnum::FlagBase;
     }
 
 	Flag::~Flag() {
+        m_FlagNode->setParent(m_FlagOldParent);
 		delete(m_FlagStatus);
 		delete(m_StartPosition);
-		delete(m_StartRotation);
+        delete(m_StartRotation);
 		delete(m_FlagNode);
```

Comment:
Old and don't know how this slipped in, but you are calling delete on a pointer that you did not allocate with new yourself. If you want to actually destroy the node, tell Irrlicht to (using the drop function I believe)